lib/dataset/aligned_dict.cpp

```diff
@@ -193,71 +199,71 @@ void Dict<Key, Value>::set(const key_type &key, mapped_type coord) {
 }
 
 template <class Key, class Value>
-void Dict<Key, Value>::erase(const key_type &key) {
+void AlignedDict<Key, Value>::erase(const key_type &key) {
   expect_writable(*this);
   scipp::expect::contains(*this, key);
   m_items.erase(key);
 }
 
 template <class Key, class Value>
-Value Dict<Key, Value>::extract(const key_type &key) {
-  auto extracted = at(key);
-  erase(key);
-  return extracted;
+Value AlignedDict<Key, Value>::extract(const key_type &key) {
+  expect_writable(*this);
+  return m_items.extract(key);
```

Comment:
Why `scipp::expect::contains` in one case but not the other? Will the exceptions be inconsistent?

Reply:
What do you mean? The function with a default value calls `AlignedDict::contains` and uses `extract` without default arg to do the extraction.

Reply:
`erase` calls `scipp::expect::contains(*this, key);` but `extract` does not. Why?

Reply:
An oversight. I changed it such that `SizedDict` no longer checks anything. It is all handled by `Dict`. But unfortunately, that means that `dict_keys_to_string` needs to be defined in `dict.h`